### ISSUE TITLE
False positive selector-no-type with ignore compounded

### ISSUE THREAD

**Issue by mehcode** [stylelint/stylelint#1201]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Seems to be a bug in `ignore: ["compounded"]`

> Which rule, if any, is this issue related to?

`selector-no-type`

> What CSS is needed to reproduce this issue?

e.g.

```css
#id input[type="file"] {
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "selector-no-type": [true, {
      "ignore": ["compounded"],
    }],
  }
}
```

> Which version of stylelint are you using?

6.2.2

> How are you running stylelint: CLI, PostCSS plugin, Node API?

PostCSS plugin

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
test.css
1:5    ⚠  Unexpected type selector (selector-no-type) [stylelint]
```

**Comment by jeddy3** [comment]
@mehcode Thanks for the report. 

It's likely something do to with the *compound selector* being part of a  *complex selector* i.e. the `#id` at the start. I think we're [mistakenly disallowing this](https://github.com/stylelint/stylelint/blob/master/src/rules/selector-no-type/__tests__/index.js#L127-L130), when it should be allowed.

Would you like to have a go at contributing a fix?

If so, I think you'll need to:

1. Remove [this test](https://github.com/stylelint/stylelint/blob/master/src/rules/selector-no-type/__tests__/index.js#L127-L130)
2. Add a test [here](https://github.com/stylelint/stylelint/blob/master/src/rules/selector-no-type/__tests__/index.js#L116)
3. Shoring up [this logic](https://github.com/stylelint/stylelint/blob/master/src/rules/selector-no-type/index.js#L49-L56)... I think [this line](https://github.com/stylelint/stylelint/blob/master/src/rules/selector-no-type/index.js#L52) looks ominous.
4. Update the changelog

The a [small section](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-bugs) in the docs about fixes, and [another](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#running-tests) about running tests.
